src/main/scala/nl.knaw.dans.easy.bagstore/server/StoresServletComponent.scala

```diff
@@ -147,6 +147,34 @@ trait StoresServletComponent {
       }
     }
 
+    head("/:bagstore/bags/:uuid/*") {
+      val bagstore = params("bagstore")
+      val uuidStr = params("uuid")
+      multiParams("splat") match {
+        case Seq(path) =>
+          bagStores.getBaseDirByShortname(bagstore)
+            .map(baseDir => ItemId.fromString(s"""$uuidStr/${ path }""")
+              .recoverWith {
+                case _: IllegalArgumentException => Failure(new IllegalArgumentException(s"Invalid UUID string: $uuidStr"))
+              }
+              .flatMap(itemId => {
+                getHeaders(itemId, baseDir)
```

Comment:
There is A LOT of duplication here. Please extract a method. The only 'real' difference is the

```scala
getHeaders(itemId, baseDir)
```

here in `HEAD` and 

```scala
debug(s"Retrieving item $itemId")
bagStores.copyToStream(itemId, request.header("Accept").flatMap(acceptToArchiveStreamType), response.outputStream, Some(baseDir))
```

in the `GET` above.

So what you can do is provide a function as an argument to that extracted function.

```scala
def foo(f: ItemId => Try[ActionResult]) = ???

foo(itemId => getHeaders(itemId, baseDir).map(headers => Ok(headers = headers))) // HEAD
foo(itemId => { // GET
    debug(s"Retrieving item $itemId")
    bagStores.copyToStream(itemId, request.header("Accept").flatMap(acceptToArchiveStreamType), response.outputStream, Some(baseDir))
        .map(_ => Ok())
})

Reply:
I'll wait till Jan has given his opinion.  It is not yet decided whether we solve the problem this way.  If this is the way we choose, then shall I optimize this piece of code.